mbed_lib.json

```diff
@@ -54,10 +70,13 @@
         "DISCO_L496AG": {
             "tx": "PB_6",
```

Comment:
Could we use STMOD_2 name ?
https://github.com/ARMmbed/mbed-os/blob/46603f831e13705d5aab4f73d52f00611e1f4d7d/targets/TARGET_STM/TARGET_STM32L4/TARGET_STM32L496xG/TARGET_DISCO_L496AG/PinNames.h#L354

Reply:
That sounds sensible to me.

Reply:
excellent idea - will make the driver configuration valid for any target with STMOD+ connector